Add `--context` option to `format` and `expand`.

BigBlueHat opened the pull request:
Mainly, I wanted to be able to test JSON documents (which completely lack an `@context` property) by generating nquads and comparing that output as I change a local context file.

```sh
jsonld format -f nquads random.json --context=context.jsonld
```

This also provides use of `-c, --context` on the `expand` command allowing for [Interpreting JSON as JSON-LD](https://www.w3.org/TR/json-ld11/#interpreting-json-as-json-ld).

BigBlueHat committed:
Add --context option to format.

BigBlueHat committed:
Add --context to expand.

BigBlueHat committed:
Add --context to lint command.

BigBlueHat committed:
Refactor cmd.command checks into _jsonLdOptions.

vorburger commented:
@BigBlueHat @davidlehn @msporny any chance you would consider merging this? I would use it on https://github.com/enola-dev/enola/pull/779! 😺

msporny commented:
@davidlehn any reason this PR is languishing? It feels fine to me. Can we merge it?

davidlehn commented:
Does this work for `compact` and `flatten`?  It's using the generic `_jsonLdOptions()` to set `options.expandContext` when `--context` is used.  But `compact` and `flatten` already used `--context` to set the param for the API call.  I'm not thinking this through all the way, but I don't think that will result in correct behavior.  This perhaps need to be something like `--expand-context` and setup for all API calls that use that option.

vorburger commented:
Actually I originally missed that this PR is specifically for adding `--context` to `format`, `lint` and `expand`; whereas I am using `flatten` which already has a `--context` flag... but you are right, it does not appear to work (on v2.0.0); if I (x2) remove [this](https://github.com/enola-dev/enola/blob/447133ceb759cf87df30dbc18e54b8606c904272/test/picasso.json#L3) `@context` (as I will be in https://github.com/enola-dev/enola/pull/796, WIP), then `npx jsonld-cli@2.0.0 flatten --context=test/picasso-context.jsonld test/picasso.json` does NOT seem to read it (nor is there an error; it seems to just ignore it). Nor does `npx jsonld-cli@2.0.0 flatten --context=picasso-context.jsonld --base=test/ test/picasso.json` work (re. #15). (With https://github.com/filip26/ld-cli it works, as used in `tools/test-jsonld/test-jsonld.bash`).

PS, just FYI: I had used `jsonld-cli` for exploring and learning JSON-LD, and am now moving on to using another implementation.

BigBlueHat commented:
> Does this work for `compact` and `flatten`? It's using the generic `_jsonLdOptions()` to set `options.expandContext` when `--context` is used. But `compact` and `flatten` already used `--context` to set the param for the API call. I'm not thinking this through all the way, but I don't think that will result in correct behavior. This perhaps need to be something like `--expand-context` and setup for all API calls that use that option.

I'd rather not have a new command line option, but since this is only for the `expand` command I'll add some restrictions to that option being set. So `--context` on `expand` will set `options.expandContext`. More accurate and avoids setting `expandContext` option on `flatten`, etc. that don't use that option.

BigBlueHat committed:
Set expandContext only for `expand` command.

BigBlueHat committed:
Use https URL in .editorconfig comment.

BigBlueHat committed:
Remove eslint-plugin-jsdoc.

JSDoc is not used in this project and the dependency was causing issues
during CI linting.

BigBlueHat committed:
Add package-lock.json.

bparth24 reviewed (approved):
LGTM.
